### ISSUE TITLE
Training pSp encoder converged too soon

### ISSUE THREAD

**Issue by kimyanna** [eladrich/pixel2style2pixel#297]
Hi,

I trained the pSp encoder on a set of 6k real->cartoon image pairs. I trained for 70k iterations after which the test/loss function converged and stopped improving. The results after 70k iterations are still too far away from a good outcome.

Below is a screenshot of Tensorboard as well as examples of latest best model and training settings.

Any recommendation on how to get a better results? What hyper-parameters might be worth changing?

Thank you for your great work!

Tensorboard after 70k iterations:
![Screen Shot 2022-10-30 at 2 18 55 PM](https://user-images.githubusercontent.com/76635807/198902310-bc2a0164-633b-46f4-a310-3ee7968c45fc.png)

Model after 70k iterations:
![0097_69000](https://user-images.githubusercontent.com/76635807/198902108-a6328ade-e746-43ab-bd60-3e016167b687.jpg)
![0087_69000](https://user-images.githubusercontent.com/76635807/198902127-45fa4589-1cc6-420f-b8e6-9e35b50b681f.jpg)
![0070_69000](https://user-images.githubusercontent.com/76635807/198902293-c838ee6e-7549-48e3-9844-de70ba568363.jpg)

Hyper-parameters / Settings:
```
{
    "batch_size": 4,
    "board_interval": 50,
    "checkpoint_path": null,
    "dataset_type": "toonify",
    "encoder_type": "GradualStyleEncoder",
    "exp_dir": "/home/ubuntu/out",
    "id_lambda": 1.0,
    "image_interval": 100,
    "input_nc": 3,
    "l2_lambda": 1.0,
    "l2_lambda_crop": 0,
    "label_nc": 0,
    "learn_in_w": false,
    "learning_rate": 0.0001,
    "lpips_lambda": 0.8,
    "lpips_lambda_crop": 0,
    "max_steps": 500000,
    "moco_lambda": 0,
    "optim_name": "ranger",
    "output_size": 1024,
    "resize_factors": null,
    "save_interval": null,
    "start_from_latent_avg": true,
    "stylegan_weights": "pretrained_models/stylegan2-ffhq-config-f.pt",
    "test_batch_size": 2,
    "test_workers": 2,
    "train_decoder": false,
    "use_wandb": false,
    "val_interval": 1000,
    "w_norm_lambda": 0.025,
    "workers": 4
}
```

**Comment by yuval-alaluf** [comment]
I think the problem you are facing here is that you are using the regular SG2 generator. When doing so, the generator can't really output cartoon images. Therefore, regardless of the parameters, you won't be able to get satisfactory results. 
You probably need to train a SG2 generator on your cartoon data.
Notice that in our real-to-toon task, we used the toon generator to get our results.

**Comment by kimyanna** [comment]
Oh yeah, that's most likely the case, I"m using a regular FFHQ SG2 generator. I'll fine-tune the generator on the cartoon images and give it another try. Thank you for your suggestion!

**Comment by kimyanna** [comment]
After fine-tuning SG2 and using it as a PSP generator (passed it to `train.py` via `--stylegan_weights=<FINE_TUNED_MODEL>.pt`) and training for 34k iterations I'm definitely getting better results than above, but it's still a little off.

Tensorboard after 34k iterations:
![Screenshot 2022-12-06 at 15-47-34 TensorBoard](https://user-images.githubusercontent.com/76635807/206050881-6eb6ffb2-9460-4637-a247-7c014783e726.png)

Results after 34k iterations:
![0097_34000_AAAA](https://user-images.githubusercontent.com/76635807/206050967-e01452a0-d1fa-44fa-81f6-a47457ae81cb.jpg)
![0087_34000_AAAA](https://user-images.githubusercontent.com/76635807/206050984-02fdd2f2-2121-49a9-9d93-b4cec030da10.jpg)
![0070_34000_AAAA](https://user-images.githubusercontent.com/76635807/206051017-72ff8a61-d5e3-478e-81bc-cf2c50b20471.jpg)
![0095_34000](https://user-images.githubusercontent.com/76635807/206051735-df9379ff-6eb1-4ac9-a33c-b89f3e88f93c.jpg)


Do you have a recommendation on what to try next to get even better outcome? All of the images are aligned and the cartoon data set (6k samples) used for fine tuning the SG2 generator and for training PSP (using image pairs) is the same.

Perhaps try training longer beyond the 34k iterations?

Thank you!

**Comment by watertianyi** [comment]
> 来要尝试什么以获得更好的结果有什

train a SG2 generator on your cartoon data，Which code is used to train the cartoon to obtain stylegan_weights?
